packages/plugin/src/Integrations/PaymentGateways/PayPal/Fields/PayPalField.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Solspace\Freeform\Integrations\PaymentGateways\PayPal\Fields;
+
+use Solspace\Freeform\Attributes\Field\Type;
+use Solspace\Freeform\Attributes\Property\Implementations\Field\FieldTransformer;
+use Solspace\Freeform\Attributes\Property\Implementations\Integrations\IntegrationTransformer;
+use Solspace\Freeform\Attributes\Property\Input;
+use Solspace\Freeform\Attributes\Property\Validators\Required;
+use Solspace\Freeform\Attributes\Property\ValueTransformer;
+use Solspace\Freeform\Attributes\Property\VisibilityFilter;
+use Solspace\Freeform\Bundles\Integrations\Providers\FormIntegrationsProvider;
+use Solspace\Freeform\Fields\AbstractField;
+use Solspace\Freeform\Fields\FieldInterface;
+use Solspace\Freeform\Fields\Interfaces\NumericInterface;
+use Solspace\Freeform\Fields\Interfaces\OptionsInterface;
+use Solspace\Freeform\Fields\Interfaces\TextInterface;
+use Solspace\Freeform\Integrations\PaymentGateways\Common\Currency\CurrencyOptionsGenerator;
+use Solspace\Freeform\Integrations\PaymentGateways\Common\PaymentFieldInterface;
+use Solspace\Freeform\Integrations\PaymentGateways\PayPal\PayPal;
+use Solspace\Freeform\Library\Attributes\Attributes;
+use Solspace\Freeform\Library\Helpers\HashHelper;
+
+#[Type(
+    name: 'PayPal Payment',
+    typeShorthand: 'paypal',
+    iconPath: __DIR__.'/../icon.svg',
+    previewTemplatePath: __DIR__.'/../Templates/paypal-field-preview.ejs',
+)]
+class PayPalField extends AbstractField implements PaymentFieldInterface
+{
+    public const AMOUNT_TYPE_FIXED = 'fixed';
+    public const AMOUNT_TYPE_DYNAMIC = 'dynamic';
+    #[Required]
```

Comment:
Needs a newline here.

```suggestion
    public const AMOUNT_TYPE_DYNAMIC = 'dynamic';
    
    #[Required]
```